egs/wsj/s5/utils/queue.pl

```diff
@@ -90,6 +98,37 @@ ()
   exit 1;
 }
 
+sub caught_signal {
+	if ($submit_flag ==1) { #captured signal after submitting jobs
+		system ("qdel $sge_job_id");
+		die "Caught a signal: $!, deleting SGE task: $sge_job_id and exiting\n";
+	}
+}
+
+sub check_status {
```

Comment:
OK, this is in response to Vijay's comment about checking the status before the job is finished....
firstly, I'm not sure whether we should do it in the same PR because the queue.pl script is easy to break.

Secondly, the way you've done it is problematic for a few reasons:
   - it would cause too much NFS traffic as it's run too frequently and on too many files (the loop it's called from only accesses one file per iteration; not all of them).  The queue.pl script has been finely tuned to be responsive enough without hogging NFS traffic. 
   - it's not ideal to spawn too many processes (tail -1), they fill up the system logs and cause other problems...  better to fseek not too far from the end of the file, read a few lines, and take the last of the lines, without invoking new processes.
   - I don't see that you have made allowances for what happens when there are left-over logs from a previous iteration.  [need to check file times.]

Overall, it may be better to make this a separate PR.